modules/lib/src/main/resources/assets/js/app/wizard/PageComponentsTreeGrid.ts

```diff
@@ -70,6 +70,15 @@ export class PageComponentsTreeGrid
         PageState.getEvents().onComponentUpdated((event: ComponentUpdatedEvent) => {
             this.updateItemByPath(event.getPath());
         });
+
+        const isRootItemEmptyChecker = this.toggleRootItemHasChildren.bind(this);
+
+        this.onLoaded(isRootItemEmptyChecker);
+        this.onDataChanged(isRootItemEmptyChecker);
+    }
+
+    private toggleRootItemHasChildren(): void {
+        this.toggleClass('empty', !this.isRootItemWithChildren());
```

Comment:
It works fine, but `empty` name for a CSS class is a) too generic, b) in this case not entirely correct. It should be `root-leaf` or something like that.